test/test_deep_merge.rb

```diff
@@ -46,6 +46,12 @@ def test_deep_merge
     # merge tests (moving from basic to more complex)
 
     # test merging an hash w/array into blank hash
+    hash_src = {'something' => true}
+    hash_dst = {'something' => false}
+    DeepMerge::deep_merge!(hash_src, hash_dst, {:preserve_unmergeables => true})
+    assert_equal({'something' => false}, hash_dst)
```

Comment:
Maybe I'm just confused by the expected behavior of this library. Are you suggesting that boolean values should be considered "unmergeable"?